### PR TITLE
feature/chats

### PR DESCRIPTION
**Pull request by Vokoloven**
In this ticked I created CRUD for chats.
You can add new, delete, or edit it.
Please review main files such as: 
1. SharedLayout here is render table with buttons
2. RowButton mapped component as button.

After successfully login open aside menu.
image

Here rendering chats: 
image

We can edit or delete it by click "More" button : 
image

Many thanks in advance!

### CODE REVIEW

**Comment by AndriiKasian** [comment]
- [ ] when we click the "New Chat" button and the modal appears, the text field shouldn't contain the previous value inside

**Comment by AndriiKasian** [comment]
- [ ]  when I was trying to create a new Chat, but my token was expired. Here is a happy pass case: before sending some request to the server you have to check the token and update it if it's needed, and after that continue to send request with the new token.

Снимок экрана 2024-04-02 в 10 55 20

File: src/App.tsx [hunk 1]
```diff
@@ -14,6 +14,7 @@ const SharedLayout = lazy(() => import("./pages/SharedLayout/SharedLayout"));
 const NotFound = lazy(() => import("./pages/NotFound/NotFound"));
 const UpworkFeed = lazy(() => import("./pages/UpworkFeed/UpworkFeed"));
 const UpworkFeedDetail = lazy(() => import("./pages/UpworkFeedDetail/UpworkFeedDetail"));
+const Chat = lazy(() => import("./pages/Chat/Chat"));
```
Comment (AndriiKasian) [inline src/App.tsx:17]: - [ ] check the conversation tab, I left comments there

**Review by AndriiKasian — CHANGES_REQUESTED** [review]

File: src/hooks/useContainerDimension.ts [hunk 2]
```diff
@@ -3,21 +3,23 @@ import {useState, useEffect} from "react";
 export const useContainerDimension = (ref: React.RefObject<HTMLElement>) => {
   const [elementHeight, setElementHeight] = useState(0);
 
-  useEffect(() => {
-    const height = () => {
-      if (ref.current) {
-        const height = ref.current.getBoundingClientRect().height;
+  const height = () => {
```
Comment (AndriiKasian) [inline src/hooks/useContainerDimension.ts:6]: - [ ] what is the main idea of this hook?

File: src/hooks/useCreateChat.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+import {AppRoutes} from "../AppRoutes";
+import CreateChat from "../components/Shared/Chat/CreateChat";
+import {useCreateChatMutation} from "../redux/api/chatsApi";
+import {STATUS_CODE} from "../redux/utils";
+import {ICreateChatRequest} from "../submodules/interfaces/dto/chat/dto/icreate-chat-request.interface";
+
+import {useModal} from "./useModal";
+
+export const useCreateChat = () => {
```
Comment (AndriiKasian) [inline src/hooks/useCreateChat.tsx:12]: - [ ] rename useCreateChat to useCreateChatModal

**Review by AndriiKasian — CHANGES_REQUESTED** [review]

File: src/components/Shared/Header.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+            />
+          </div>
+        </div>
+        <div className={classnames(styles.sectionHeaderBoxOuter, {[`${styles.collapsed}`]: collapsed})}>
+          <div className={styles.sectionHeaderBoxInner}>
+            <Button
+              size={ButtonSize.Small}
+              text="New Chat"
+              iconBefore={IconAppName.Cross}
+              // onClick={() => openModal()}
```
[the remark this answers is not in the document]
Comment (Vokoloven) [reply src/components/Shared/Header.tsx:40]: Yes we this button will shown when device resolution as tablet and less.
image

**Review by AndriiKasian — APPROVED** [review]

**Review by AndriiKasian — DISMISSED** [review]

**Pull request merged**